packages/@aws-cdk/cdk/lib/aspects/tag-aspect.ts

```diff
@@ -0,0 +1,120 @@
+import { ITaggable, Resource } from '../cloudformation/resource';
+import { IConstruct } from '../core/construct';
+import { Aspect } from './aspect';
+
+export interface TagBaseProps extends TagProperties {
+  /**
+   * The value of the tag
+   */
+  value?: string;
+}
+
+export interface TagProperties {
+  /**
+   * This applies specifically to AutoScalingGroup PropagateAtLaunch
+   */
+  applyToLaunchInstances?: boolean;
+
+  /**
+   * An array of Resource Types that will receive this tag
+   */
+  include?: string[];
+
+  /**
+   * An array of Resource Types that will not receive this tag
+   */
+  exclude?: string[];
+}
+
+/**
+ * The common functionality for Tag and Remove Tag Aspects
+ */
+export abstract class TagBase extends Aspect {
+
+  /**
+   * Test if the construct is a CloudFormation Resource
+   */
+  public static isResource(resource: any): resource is Resource {
+    return resource.resourceType !== undefined;
+  }
+
+  /**
+   * The ``taggable`` type for these aspects
+   */
+  public readonly type: string = 'taggable';
+
+  /**
+   * The string key for the tag
+   */
+  public readonly key: string;
+
+  /**
+   * The string value of the tag
+   */
+  public readonly value?: string;
```

Comment:
Value is only needed for Tag and not for RemoveTag so why is it in TagBase?

Reply:
yep - a mistake for sure, will fix